src/scene/vcPolyModelNode.cpp

```diff
@@ -247,3 +229,26 @@ udDouble3 vcPolyModelNode::GetLocalSpacePivot()
 
   return m_pModel->origin;
 }
+
+void vcPolyModelNode::LoadTransforms(const udGeoZone &zone)
+{
+  if (m_pNode->geomCount != 0)
+  {
+    udDouble3 *pPosition = nullptr;
+    udDouble3 scale;
+    udDouble3 euler;
+
+    vcProject_FetchNodeGeometryAsCartesian(m_pProject, m_pNode, zone, &pPosition, nullptr);
+
+    udProjectNode_GetMetadataDouble(m_pNode, "transform.rotation.y", &euler.x, 0.0);
+    udProjectNode_GetMetadataDouble(m_pNode, "transform.rotation.p", &euler.y, 0.0);
+    udProjectNode_GetMetadataDouble(m_pNode, "transform.rotation.r", &euler.z, 0.0);
+    udProjectNode_GetMetadataDouble(m_pNode, "transform.scale.x", &scale.x, 1.0);
+    udProjectNode_GetMetadataDouble(m_pNode, "transform.scale.y", &scale.y, 1.0);
+    udProjectNode_GetMetadataDouble(m_pNode, "transform.scale.z", &scale.z, 1.0);
+
+    m_matrix = udDouble4x4::rotationYPR(UD_DEG2RAD(euler), *pPosition) * udDouble4x4::scaleNonUniform(scale);
+
+    udFree(pPosition);
+  }
+}
```

Comment:
This didn't need to be a new function if its only called from one place and that function is now one line